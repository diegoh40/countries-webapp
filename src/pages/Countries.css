.bg-1 {
    background-color: rgb(195, 0, 255);
   
  }

  .container {
    display: grid;
    grid-template-columns: auto auto;
    /* background-color: rgb(195, 0, 255); */
    padding: 10px 0 10px 0;
  }

  .bg-success {
    --bs-bg-opacity: 1;
    background-color: rgba(var(--bs-success-rgb), var(--bs-bg-opacity)) !important;
  }