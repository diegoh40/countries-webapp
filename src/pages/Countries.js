import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { showCountries, selectCountry, showCountriesAll } from '../redux/Home/home';
import './Countries.css';
import CountryTable from './tableCountries';

const Countries = () => {
  const countries = useSelector((state) => state.countries);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(showCountriesAll());
  }, []);

  const handleContinents = (event) => {
    dispatch(showCountries(event.target.value));
  };

  const handleCountry = (id) => {
    dispatch(selectCountry(id));
  };



  const worldContinents = [
    { name: 'Africa', id: 1 },
    { name: 'Asia', id: 2 },
    { name: 'Europe', id: 3 },
    { name: 'South America', id: 4 },
    { name: 'North America', id: 5 },
    { name: 'Oceania', id: 6 },
  ];

  return (
    <div className=" bg-danger-subtle" >

      <div className="select">
        <h3>Select a Continent</h3>
        <div className="d-flex flex-wrap">
          {worldContinents.map((el) => (
            <button
              key={el.id}
              onClick={handleContinents}
              value={el.name} type="button"
              className="btn btn-primary ms-2 mb-2 px-3"
              aria-hidden="true">
              {el.name}
            </button>
          ))}
        </div>
      </div>
      <br />
    
      <div className="col  items-center">
            <h1>Countries List</h1>
      <div className="m-5">
      <CountryTable countries={countries} handleCountryClick={handleCountry} />

      </div>
    </div>

    </div>
  );
};

export default Countries;
