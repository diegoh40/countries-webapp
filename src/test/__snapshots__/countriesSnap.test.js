import { Provider } from 'react-redux';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
// eslint-disable-next-line import/extensions
import store from '../redux/configureStore';
// eslint-disable-next-line import/extensions
import Countries from '../pages/Countries';

test('Countries snapshot', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <Switch>
          <Route>
            <Countries />
          </Route>
        </Switch>
      </BrowserRouter>
    </Provider>,
  );

  expect(tree).toMatchSnapshot();
});
