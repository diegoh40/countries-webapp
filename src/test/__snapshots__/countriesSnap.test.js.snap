// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <select>
            <option
              value="Africa"
            >
              Africa
            </option>
            <option
              value="Asia"
            >
              Asia
            </option>
            <option
              value="South America"
            >
              South America
            </option>
            <option
              value="North America"
            >
              North America
            </option>
            <option
              value="Europe"
            >
              Europe
            </option>
          </select>
        </form>
        <div
          class="countries"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <select>
          <option
            value="Africa"
          >
            Africa
          </option>
          <option
            value="Asia"
          >
            Asia
          </option>
          <option
            value="South America"
          >
            South America
          </option>
          <option
            value="North America"
          >
            North America
          </option>
          <option
            value="Europe"
          >
            Europe
          </option>
        </select>
      </form>
      <div
        class="countries"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
